runtime/main.cpp

```diff
@@ -64,7 +67,17 @@ class cancellable_sleep {
     std::atomic<bool> _m_terminate{false};
 };
 
+void spdlogger(std::string name, std::string log_level = "off") {
+    std::vector<spdlog::sink_ptr> sinks;
+    sinks.push_back(std::make_shared<spdlog::sinks::stdout_color_sink_mt>());
+    sinks.push_back(std::make_shared<spdlog::sinks::basic_file_sink_mt>("logs/" + name + ".log"));
+    auto logger = std::make_shared<spdlog::logger>(name, begin(sinks), end(sinks));
+    logger->set_level(spdlog::level::from_str(log_level));
+    spdlog::register_logger(logger);
+}
+
 int main(int argc, char* const* argv) {
+    spdlogger("illixr", ILLIXR::getenv_or("ILLIXR_LOG_LEVEL", "off"));
```

Comment:
I like an opt-in system, but I'm also not a plugin developer. @Jebbly @wsherman64 - want to chime in on this?

Reply:
@shg8 - are you thinking that the default state should be the most verbose?

Reply:
I changed the log_level parameter to be a const char * instead of a std::string. This allowed me to use std::getenv instead of ILLIXR::getenv_or, which required redundantly setting the default log level. This was also confusing and effectively wrote over the default parameter. This also allows for a quick check on if log_level is NULL and if so to set to 'warn'. log_level will be NULL if either the <PLUGIN_NAME>_LOG_LEVEL env var is not exported or exported as nothing, i.e. either:
# set to nothing
export PLUGIN_NAME_LOG_LEVEL=
# commented out
# export PLUGIN_NAME_LOG_LEVEL=debug